src/lib.rs

```diff
@@ -24,7 +30,7 @@ pub trait Pattern<'s> {
 macro_rules! impl_pattern {
 	($ty:ty $(where $($bound:tt)*)?) => {
 		impl<'s $(, $($bound)*)?> Pattern<'s> for $ty {
-			type MatchIndices = std::str::MatchIndices<'s, Self>;
+			type MatchIndices = core::str::MatchIndices<'s, Self>;
```

Comment:
Do references in doc comments need to be updated as well? (I don't know how / whether anyone builds docs with `no_std`)

Reply:
Considering core's docs also just refer to std, I didn't bother changing them. Idk, it's a bit weird.

Reply:
I've changed the extern crate reference to `extern crate alloc as std` - it seems less confusing and more consistent that way and allows to keep all `std::` references as-is.